src/test/java/seedu/address/logic/commands/AddAttendanceCommandTest.java

```diff
@@ -0,0 +1,236 @@
+    }
+
+    @Test
+    public void constructor_nullIndexes_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () ->
+                new AddAttendanceCommand(null, INDEX_FIRST_ITEM, INDEX_FIRST_ITEM,
+                        VALID_WEEK, VALID_ATTENDANCE));
+        assertThrows(NullPointerException.class, () ->
+                new AddAttendanceCommand(INDEX_FIRST_ITEM, null, INDEX_FIRST_ITEM,
+                        VALID_WEEK, VALID_ATTENDANCE));
+        assertThrows(NullPointerException.class, () ->
+                new AddAttendanceCommand(INDEX_FIRST_ITEM, INDEX_FIRST_ITEM, null,
+                        VALID_WEEK, VALID_ATTENDANCE));
+    }
+
+    @Test
+    public void constructor_nullWeek_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () ->
+                new AddAttendanceCommand(INDEX_FIRST_ITEM, INDEX_FIRST_ITEM, INDEX_FIRST_ITEM,
+                        null, VALID_ATTENDANCE));
+    }
+
+    @Test
+    public void execute_success() {
+        Index moduleClassIndex = INDEX_FIRST_ITEM;
+        Index lessonIndex = INDEX_FIRST_ITEM;
+        Index studentIndex = INDEX_FIRST_ITEM;
+
+        ModuleClass moduleClass = model.getFilteredModuleClassList().get(moduleClassIndex.getZeroBased());
+        Student student = model.getFilteredStudentList().get(studentIndex.getZeroBased());
+        Lesson lesson = moduleClass.getLessons().get(lessonIndex.getZeroBased());
+        AttendanceRecord attendanceRecord = new AttendanceRecord(Map.of(student.getUuid(), VALID_ATTENDANCE));
+        List<AttendanceRecord> attendanceRecords =
+                new ArrayList<>(lesson.getAttendanceRecordList().getAttendanceRecordList());
+        attendanceRecords.set(VALID_WEEK.getZeroBasedWeekIndex(), attendanceRecord);
+        Lesson modifiedLesson = new LessonBuilder()
+                .withStartTime(lesson.getStartTime())
+                .withEndTime(lesson.getEndTime())
+                .withDay(lesson.getDay())
+                .withVenue(lesson.getVenue().venue)
+                .withAttendanceRecordList(new AttendanceRecordList(attendanceRecords))
+                .build();
+        ModuleClass modifiedModuleClass = manualReplaceLessonToModuleClass(moduleClass, lesson, modifiedLesson);
+
+        String expectedMessage =
+                String.format(AddAttendanceCommand.MESSAGE_SUCCESS, student.getName(), VALID_WEEK, VALID_ATTENDANCE);
+        Model expectedModel = new ModelManager(model.getTutorsPet(), new UserPrefs());
+        expectedModel.setModuleClass(moduleClass, modifiedModuleClass);
+        expectedModel.commit(expectedMessage);
+
+        AddAttendanceCommand addAttendanceCommand =
+                new AddAttendanceCommand(moduleClassIndex, lessonIndex, studentIndex, VALID_WEEK, VALID_ATTENDANCE);
+
+        assertCommandSuccess(addAttendanceCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_existingAttendance_failure() {
```

Comment:
Perhaps we should have a test for the case where the student is not a member of the class?